src/main/kotlin/com/terraformation/backend/tracking/db/DeliveryStore.kt

```diff
@@ -95,8 +97,112 @@ class DeliveryStore(
     }
   }
 
+  fun reassignDelivery(deliveryId: DeliveryId, reassignments: List<Reassignment>) {
+    requirePermissions { updateDelivery(deliveryId) }
+
+    val now = clock.instant()
+    val userId = currentUser().userId
+
+    val plantingSiteId = getPlantingSiteId(deliveryId)
+    val originalPlantingIds = reassignments.map { it.fromPlantingId }
+    val originalPlantings =
+        plantingsDao.fetchById(*originalPlantingIds.toTypedArray()).associateBy { it.id!! }
+    val deliveryPlantings = plantingsDao.fetchByDeliveryId(deliveryId)
+
+    val newPlantings =
+        reassignments.flatMap { reassignment ->
+          val fromPlantingId = reassignment.fromPlantingId
+          val originalPlanting =
+              originalPlantings[fromPlantingId] ?: throw PlantingNotFoundException(fromPlantingId)
+          val speciesId = originalPlanting.speciesId!!
+
+          if (originalPlanting.deliveryId != deliveryId) {
+            throw PlantingNotFoundException(fromPlantingId)
```

Comment:
if both deliveries are within the same org, thoughts on sharing more info .. similar to `CrossOrganizationNurseryTransferNotAllowedException`

Reply:
Added a more specific exception for this, which also prompted me to add a permission check for the original planting (updated the previous PR in the stack to include that).